### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by aberonja**
Frontend initial setup and basic routing

### CODE REVIEW

File: src/main/frontend/src/App.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, {Component} from 'react';
+import './App.css';
+import Login from './pages/Login';
+import Dashboard from './pages/Control';
+import {Switch, Route, BrowserRouter, Redirect} from 'react-router-dom';
+
+class App extends Component {
```
Comment (iscekic) [inline src/main/frontend/src/App.js]: No need for `export default` in separate line.

File: src/main/frontend/src/components/ExecomCalendar.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React, {Component} from 'react';
+
+export default class ExecomCalendar extends Component {
+  render() {
+    return (
+      <div>
```
Comment (iscekic) [inline src/main/frontend/src/components/ExecomCalendar.js]: Is this div actually needed for layout purposes? If no, use a `Fragment`.

File: src/main/frontend/src/pages/Control.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import Leave from '../components/Leave';
+import LeaveHistory from '../components/LeaveHistory';
+import TeamCalendar from '../components/TeamCalendar';
+import ExecomCalendar from '../components/ExecomCalendar';
+
+export default class Control extends Component {
+  render() {
+    return (
+      <div>
+        <NavLink className="link" to="/leave">
```
Comment (iscekic) [inline src/main/frontend/src/pages/Control.js]: Can these `NavLinks` be generated programatically?

File: src/main/frontend/src/pages/Login.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React, {Component} from 'react';
+import {NavLink} from 'react-router-dom';
+
+export default class Login extends Component {
+  render() {
+    return (
+      <div>
+        <div className="background">
```
Comment (iscekic) [inline src/main/frontend/src/pages/Login.js]: These class names seem fishy. This also looks like a good candidate for a separate component.

@aberonja @dusanfle Let's use `styled-components` on this project.

https://www.styled-components.com/
https://github.com/timarney/react-app-rewired

File: src/main/frontend/src/scss/_main.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+@import 'base/variables';
```
Comment (iscekic) [inline src/main/frontend/src/scss/_main.scss]: Delete all `(s)css` files as we will use `styled-components`.

Comment (iscekic) [reply src/main/frontend/src/scss/_main.scss]: If we still need a css reset use one from `npm`.

File: src/main/frontend/package.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "hawaii-execom",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "husky": "^0.14.3",
```
[the remark this answers is not in the document]
Comment (iscekic) [reply src/main/frontend/package.json]: Rule of thumb: `dependencies` end up in the bundle, `dev-dependencies` don't. 😄

File: src/main/frontend/src/components/ExecomCalendar.js (hunk 2 above)
Comment (iscekic) [reply src/main/frontend/src/components/ExecomCalendar.js]: Same for other occurences.

File: src/main/frontend/package.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "hawaii-execom",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "react": "^16.4.0",
+    "react-dom": "^16.4.0",
+    "react-form": "^3.5.5",
```
Comment (iscekic) [inline src/main/frontend/package.json]: Remove this dependency.

File: src/main/frontend/package.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    "styled-components": "^3.3.2"
+  },
+  "lint-staged": {
+    "src/**/*.{js, jsx, json}": [
+      "prettier --single-quote --write",
+      "git add"
+    ]
+  },
+  "scripts": {
+    "start": "react-scripts start",
```
Comment (iscekic) [inline src/main/frontend/package.json]: We will probably need to use `react-app-rewired` for production builds. Please create an issue so we don't forget.

File: src/main/frontend/src/App.scss [hunk 9]
```diff
@@ -0,0 +1 @@
+@import 'scss/main';
```
Comment (iscekic) [inline src/main/frontend/src/App.scss]: Delete this file.

File: src/main/frontend/src/components/Link.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import { NavLink } from 'react-router-dom';
+
+const Linky = ({ className, children, url }) => (
```
Comment (iscekic) [inline src/main/frontend/src/components/Link.js]: Linky? 😋

File: src/main/frontend/src/components/Link.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  <NavLink to={url} className={className}>
+    {children}
+  </NavLink>
+);
+
+const StyledLink = styled(Linky)`
```
Comment (iscekic) [inline src/main/frontend/src/components/Link.js]: No need to create `Linky` and `Link`. Just do:

```js
export default styled(NavLink)`...`;
```

Comment (iscekic) [reply src/main/frontend/src/components/Link.js]: I'm also not sure if we should even style `NavLink`. We can use it as a wrapper and have the styles in the child component - what do you think?

File: src/main/frontend/src/index.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import '../node_modules/reset-css/reset.css';
```
Comment (iscekic) [inline src/main/frontend/src/index.js]: `import 'reset-css/reset.css'` didn't work?

File: src/main/frontend/src/index.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+import App from './App';
+import registerServiceWorker from './registerServiceWorker';
+import { injectGlobal } from 'styled-components';
+
+ReactDOM.render(<App />, document.getElementById('root'));
+registerServiceWorker();
+
+injectGlobal`
```
Comment (iscekic) [inline src/main/frontend/src/index.js]: Move this to another file.

File: src/main/frontend/src/pages/Control.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+import Link from '../components/Link';
+
+const navLinks = [
+  { url: 'leave', name: 'Leave' },
+  { url: 'leave-history', name: 'Leave history' },
+  { url: 'team-calendar', name: 'Team calendar' },
+  { url: 'execom-calendar', name: 'Execom calendar' }
+];
+
+export default class Control extends Component {
```
Comment (iscekic) [inline src/main/frontend/src/pages/Control.js]: `Control` isn't a very good name. `Navbar`? Or something better. 😄

File: src/main/frontend/src/pages/Control.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+  render() {
+    const linkItems = navLinks.map(navLink => (
```
Comment (iscekic) [inline src/main/frontend/src/pages/Control.js]: We can `map` directly in the definition of `navLinks`:

```js
const navLinks = [...].map(...);
```

File: src/main/frontend/src/pages/Login.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
+import React, { Component, Fragment } from 'react';
+import styled from 'styled-components';
+import Link from '../components/Link';
+
+const Title = styled.h1`
```
Comment (iscekic) [inline src/main/frontend/src/pages/Login.js]: If this component will be used in another place somewhere, extract it to its own file.

File: src/main/frontend/package.json (hunk 7 above)
Comment (iscekic) [reply src/main/frontend/package.json]: Also remember to rebuild `yarn.lock`.

File: src/main/frontend/package.json (hunk 8 above)
Comment (dusanfle) [reply src/main/frontend/package.json]: Issue is created, #7

File: src/main/frontend/package.json [hunk 17]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "hawaii-execom",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "react": "^16.4.0",
+    "react-dom": "^16.4.0",
+    "react-redux": "^5.0.7",
+    "react-router-dom": "^4.2.2",
+    "react-scripts": "1.1.4",
```
Comment (iscekic) [inline src/main/frontend/package.json]: This is a `dev-dependency`.

File: src/main/frontend/src/components/StyledLink.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+export const StyledLink = styled.span`
+  color: ${props => (props.color ? props.color : 'white')};
```
Comment (iscekic) [inline src/main/frontend/src/components/StyledLink.js]: Let's use `styled-tools`. 😄 

https://github.com/diegohaz/styled-tools

File: src/main/frontend/src/styles/BaseStyles.js [hunk 19]
```diff
@@ -0,0 +1,3 @@
+/* 
+File for the global css styles
```
Comment (iscekic) [inline src/main/frontend/src/styles/BaseStyles.js]: Why is this file here? It should probably be removed and then added when needed.

File: src/main/frontend/src/components/StyledLink.js (hunk 18 above)
Comment (iscekic) [reply src/main/frontend/src/components/StyledLink.js]: The equivalent with `styled-tools` would be:

```
color: ${prop('color', 'white')};
```
